src/mlpack/core/data/string_encoding_policies/tf_idf_encoding_policy.hpp

```diff
@@ -65,6 +72,12 @@ class TfIdfEncodingPolicy
                          size_t dictionarySize)
   {
     output.resize(datasetSize, std::vector<OutputType> (dictionarySize,0));
+    std::cout<<"std::vector<char> v;\n";
+    for(auto it=idfdict.begin();it!=idfdict.end();it++)
+    {
+      std::cout<<it->first<<" "<<it->second<<"\n";
+    }
+    std::cout<<"\n";
```

Comment:
![Screenshot from 2019-08-05 22-42-38](https://user-images.githubusercontent.com/21286134/62482423-b63d5a00-b7d2-11e9-863d-91a683b76a44.png)

Am I missing Something ?, The tfidf Values Are changing for same dataset, I don't know how !

Reply:
Did you run the `Encode()` function sequentially 2 times? Probably you didn't reset the token counters.

Reply:
@lozhnikov  Yes, The issue was with not reseting the token counters, Since they were static data members !